numcodecs/jenkins.pyx

```diff
@@ -0,0 +1,374 @@
+
+    This is reversible, so any information in (a,b,c) before mix() is
+    still in (a,b,c) after mix().
+
+    If four pairs of (a,b,c) inputs are run through mix(), or through
+    mix() in reverse, there are at least 32 bits of the output that
+    are sometimes the same for one pair and different for another pair.
+    This was tested for:
+    * pairs that differed by one bit, by two bits, in any combination
+      of top bits of (a,b,c), or in any combination of bottom bits of
+      (a,b,c).
+    * "differ" is defined as +, -, ^, or ~^.  For + and -, I transformed
+      the output delta to a Gray code (a^(a>>1)) so a string of 1's (as
+      is commonly produced by subtraction) look like a single 1-bit
+      difference.
+    * the base values were pseudorandom, all zero but one bit set, or 
+      all zero plus a counter that starts at zero.
+
+    Some k values for my "a-=c; a^=rot(c,k); c+=b;" arrangement that
+    satisfy this are
+        4  6  8 16 19  4
+        9 15  3 18 27 15
+       14  9  3  7 17  3
+    Well, "9 15 3 18 27 15" didn't quite get 32 bits diffing
+    for "differ" defined as + with a one-bit base and a two-bit delta.  I
+    used http://burtleburtle.net/bob/hash/avalanche.html to choose 
+    the operations, constants, and arrangements of the variables.
+
+    This does not achieve avalanche.  There are input bits of (a,b,c)
+    that fail to affect some output bits of (a,b,c), especially of a.  The
+    most thoroughly mixed value is c, but it doesn't really even achieve
+    avalanche in c.
+
+    This allows some parallelism.  Read-after-writes are good at doubling
+    the number of bits affected, so the goal of mixing pulls in the opposite
+    direction as the goal of parallelism.  I did what I could.  Rotates
+    seem to cost as much as shifts on every machine I could lay my hands
+    on, and rotates are much kinder to the top and bottom bits, so I used
+    rotates.
+    """
+    a -= c
+    a ^= _jenkins_lookup3_rot(c, 4)
+    c += b
+    b -= a
+    b ^= _jenkins_lookup3_rot(a, 6)
+    a += c
+    c -= b
+    c ^= _jenkins_lookup3_rot(b, 8)
+    b += a
+    a -= c
+    a ^= _jenkins_lookup3_rot(c, 16)
+    c += b
+    b -= a
+    b ^= _jenkins_lookup3_rot(a, 19)
+    a += c
+    c -= b
+    c ^= _jenkins_lookup3_rot(b, 4)
+    b += a
+    return a, b, c
+
+class JenkinsLookup3(Checksum32):
+    """Bob Jenkin's lookup3 checksum with 32-bit output
+
+    This is the HDF5 implementation.
+    https://github.com/HDFGroup/hdf5/blob/577c192518598c7e2945683655feffcdbdf5a91b/src/H5checksum.c#L378-L472
+
+    With this codec, the checksum is concatenated on the end of the data
+    bytes when encoded. At decode time, the checksum is performed on
+    the data portion and compared with the four-byte checksum, raising
+    RuntimeError if inconsistent.
+    """
+
+    checksum = jenkins_lookup3
+    codec_id = "bob_jenkins_lookup3"
+
+    def encode(self, buf, initval=0):
```

Comment:
Indeed, initval will never be non-default like this - if you want to use it at all, it should be in `__init__` and stored as an attribute

Reply:
Resolved in [296e4ea](https://github.com/zarr-developers/numcodecs/pull/446/commits/296e4eaf809a9d73c0ee3b295a99d64ea023cb79)